packages/core/core/src/types.js

```diff
@@ -86,14 +85,27 @@ export type Target = {|
   source?: FilePath | Array<FilePath>,
 |};
 
+export const SpecifierType = {
+  esm: 0,
+  commonjs: 1,
+  url: 2,
+  custom: 3,
+};
+
+export const Priority = {
+  sync: 0,
+  parallel: 1,
+  lazy: 2,
+};
+
 export type Dependency = {|
   id: string,
-  moduleSpecifier: ModuleSpecifier,
-  isAsync: boolean,
-  isEntry: ?boolean,
+  specifier: DependencySpecifier,
+  specifierType: $Values<typeof SpecifierType>,
+  priority: $Values<typeof Priority>,
```

Comment:
These are stored as numbers rather than strings for better compression. A further optimization would be to combine `specifierType`, `priority`, `needsStableName`, `isEntry`, and `isOptional` into a single bitfield.